src/link/edit.js

```diff
@@ -69,15 +76,44 @@ const Edit = ( {
 		newValue.start = newValue.end;
 		newValue.activeFormats = [];
 		onChange( newValue );
+		setPreviewTitle(title)
 		stopAddingPreview();
+		startViewingPreview();
 		onFocus();
 	};
 
 	const removeAttributes = () => {
-		onChange( removeFormat( value, formatType ) );
+		if (previewTitle && !activePreview) {
+			const start = value.end - previewTitle.length
+			onChange( removeFormat( value, formatType, start, value.end ) );
+		} else {
+			onChange( removeFormat( value, formatType ) );
+		}
 		stopAddingPreview();
+		stopViewingPreview();
 	};
 
+	const goToEdit = () => {
+		stopViewingPreview();
+		startAddingPreview();
+	}
+	
+	const onClosePreview = () => {
+		stopViewingPreview()
+		if (activePreview && previewTitle === activePreview.attributes.title) {
```

Comment:
the `useEffect` in line 109 set the preview title, so wondering what's the purpose of the condition of `previewTitle === activePreview.attributes.title` here

Reply:
Hmm actually you're right, that condition is not needed so will remove it. It was a leftover from something I was trying earlier.